src/tokens.js

```diff
@@ -271,101 +272,288 @@ const tokens = {
-      // Output3: new melt authority
-      const tokenOutput3 = {'address': address, 'value': TOKEN_MELT_MASK, 'tokenData': tokenData};
-      // Create new data
-      let newTxData = {'inputs': [newInput], 'outputs': [tokenOutput1, tokenOutput2, tokenOutput3], 'tokens': [token]};
-      try {
-        // Get new data to sign
-        const newDataToSign = transaction.dataToSign(newTxData);
-        // Sign mint tx
-        newTxData = transaction.signTx(newTxData, newDataToSign, pin);
-        // Assemble tx and send to backend
-        transaction.completeTx(newTxData);
-        const newTxBytes = transaction.txToBytes(newTxData);
-        const newTxHex = util.buffer.bufferToHex(newTxBytes);
-        walletApi.sendTokens(newTxHex, (response) => {
-          if (response.success) {
-            resolve();
-          } else {
-            reject(response.message);
+  mintTokens(txID, index, addressSpent, token, address, amount, pin, createAnotherMint, createMelt) {
+    // Get mint data
+    let newTxData = this.getMintData(txID, index, addressSpent, token, address, amount, createAnotherMint, createMelt);
+    return transaction.sendTransaction(newTxData, pin);
+  },
+
+  /**
+   * Generate melt data
+   *
+   * @param {string} txID Hash of the transaction to be used to melt tokens
+   * @param {number} index Index of the output being spent
+   * @param {string} addressSpent Address of the output being spent
+   * @param {string} token Token uid to be melted
+   * @param {number} amount Amount of the token to be melted
+   * @param {boolean} createAnotherMelt If should create another melt output after spending this one
+   *
+   * @return {Object} Melt data {'inputs', 'outputs', 'tokens'}
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  getMeltData(txID, index, addressSpent, token, amount, createAnotherMelt) {
+    console.log('GET MELT DATA', createAnotherMelt);
+    // Get inputs that sum at least the amount requested to melt
+    const result = this.getMeltInputs(amount, token);
+
+    // Can't find inputs to this amount
+    if (result === null) return null;
+
+    // First adding authority input with MELT capability that will be spent
+    const authorityInput = {'tx_id': txID, 'index': index, 'token': token, 'address': addressSpent};
+    // Then adding the inputs with the amounts
+    const inputs = [authorityInput, ...result.inputs];
+    const outputs = [];
+    const tokens = [token];
+
+    if (result.inputsAmount > amount) {
+      // Need to create change output
+      const newAddress = wallet.getAddressToUse();
+      outputs.push({'address': newAddress, 'value': result.inputsAmount - amount, 'tokenData': 1});
+    }
+
+    if (createAnotherMelt) {
+      // New melt authority for this wallet
+      const newAddress = wallet.getAddressToUse();
+      outputs.push({'address': newAddress, 'value': TOKEN_MELT_MASK, 'tokenData': AUTHORITY_TOKEN_DATA});
```

Comment:
Would it be `AUTHORITY_TOKEN_DATA | token_id`?

Reply:
No, actually the `AUTHORITY_TOKEN_DATA` is in https://github.com/HathorNetwork/hathor-wallet-lib/blob/feat/token-management/src/constants.js#L121.

This value is the `TOKEN_AUTHORITY_MASK` with 1 (that is the index of the token we are managing - mint, melt, delegate, destroy). We are assuming at this point that those transactions that execute those actions are only for one token

Reply:
I checked the constants file, and this assumption is not there. I think we should let `AUTHORITY_TOKEN_DATA = 0b10000000`, and we should create and use a new constant called `AUTHORITY_TOKEN_DATA_FIRST` (or something like that).

Reply:
We already have `TOKEN_AUTHORITY_MASK = 0b10000000`.
Maybe I should just rename the variable to `AUTHORITY_TOKEN_DATA_FIRST ` and explain this in the comments. What do you think?

Reply:
Maybe just improve its comment in the constants file.

I'd suggest something like "Token data for an authority output of the first token in a transaction. As most transactions with authority outputs have only one token, it may be directly used, as a shortcut."

And maybe change `0b10000001` to `TOKEN_AUTHORITY_MASK | 1`. What do you think?

Reply:
Sounds good! Changed it